### PR TITLE
Cloudwatch: Pass label in deep link

### LINKED ISSUE #49158: CloudWatch: Use label field in deep link
In case the dynamic label feature is enabled and the label field is set, it should be included in the deep link from Grafana to the CloudWatch console.

part of https://github.com/grafana/grafana/issues/48434

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

In case the dynamic label feature is enabled and the label field is set, it should be included in the deep link from Grafana to the CloudWatch console.

**Which issue(s) this PR fixes**:

Fixes #49158
Part of https://github.com/grafana/grafana/issues/48434

### CODE REVIEW

File: pkg/tsdb/cloudwatch/cloudwatch_query_test.go [hunk 1]
```diff
@@ -9,27 +9,107 @@ import (
...
+					"InstanceId": {"i-12345678"},
+				},
+				MetricQueryType:  MetricQueryTypeSearch,
+				MetricEditorMode: MetricEditorModeBuilder,
+			},
+			featureEnabled: true,
+			// decoded: label: ${PROP('Namespace')}
+			linkContainsSubstring: "label%22%3A%22%24%7BPROP%28%27Namespace%27%29%7D%22%7D%5D%5D%7D",
+		},
+			"does not include label in case dynamic label is diabled": {
```
Comment (sarahzinger) [inline pkg/tsdb/cloudwatch/cloudwatch_query_test.go:60]:
```suggestion
			"does not include label in case dynamic label is disabled": {
```

**Review by sarahzinger — APPROVED** [review]
🎉 probably good to get another set of eyes, since my go is not great, but it seems good to me!

File: pkg/tsdb/cloudwatch/cloudwatch_query_test.go [hunk 2]
```diff
@@ -9,27 +9,107 @@ import (
...
+					Id:         "id1",
+					MatchExact: true,
+					Label:      "${PROP('Namespace')}",
+					Dimensions: map[string][]string{
+						"InstanceId": {"i-12345678"},
+					},
+					MetricQueryType:  MetricQueryTypeSearch,
+					MetricEditorMode: MetricEditorModeBuilder,
+				},
+				featureEnabled: true,
```
Comment (iwysiu) [inline pkg/tsdb/cloudwatch/cloudwatch_query_test.go:76]: Based on the test name, should this be `featureEnabled: false`?

File: pkg/tsdb/cloudwatch/cloudwatch_query_test.go [hunk 3]
```diff
@@ -9,27 +9,107 @@ import (
...
+				// decoded: label: ${PROP('Namespace')}
+				linkContainsSubstring: "label%22%3A%22%24%7BPROP%28%27Namespace%27%29%7D%22%7D%5D%5D%7D",
+			},
+			"includes label in case dynamic label is enabled and its a math expression query": {
```
Comment (iwysiu) [inline pkg/tsdb/cloudwatch/cloudwatch_query_test.go:80]: I think this is exactly the same test as the first test case

**Review by iwysiu — COMMENTED** [review]
I think the table tests are accidentally identical.

File: pkg/tsdb/cloudwatch/cloudwatch_query_test.go (hunk 3 above)
Comment (sunker) [reply pkg/tsdb/cloudwatch/cloudwatch_query_test.go]: Yeah thanks for finding this! Actually table tests were not suitable here so did some refactoring here.

**Pull request merged**